src/Development/IDE/Core/Rules.hs

```diff
@@ -570,14 +585,20 @@ typeCheckRuleDefinition hsc pm isFoi source = do
         case isFoi of
           IsFOI Modified -> return (diags, Just tcm)
```

Comment:
how come we don't compute HIE information for files of interest?

Reply:
We do, in the `GetHieAst` rule. `Typecheck` only computes HIE information if it is writing out the ifaces. If we aren't doing that, typecheck will not compute it. `GetHieAst` depends on `Typecheck`, and will reuse the information if `Typecheck` already computed it, otherwise it will compute it itself.